zboss/CMakeLists.txt

```diff
@@ -0,0 +1,51 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+include(${NRFXLIB_DIR}/common.cmake)
+
+# Common includes, not related to ZBOSS source code.
+zephyr_include_directories(src/zb_error)
+zephyr_include_directories(include/addons)
+
+# Source files
+zephyr_sources(src/zb_error/zb_error_to_string.c)
```

Comment:
The reason is quite simple:
The zephyr_library_* extension is mentioned and documented only in the CMake source code (at least - I cannot find it elsewhere). This makes it hard to find, understand and use. The only examples you may find is to go through other modules and try to find something similar to your needs. It would be great if you can point me to the tutorial on how to use zephyr_library_* extensions.
On the other hand - if you try to use "raw" CMake commands (fall-back to what you can find on the Internet), it will fail to link to the default zephyr target.

This should not be a surprise that, becasue zephyr_sources/include_directories/link_libraries functions are easiest to use, they are the first choice for a developer with small experience in Zephyr's extensions to CMake files.

From what I see - our project needs to define at least three libraries. I hope that this approach will meet your requirements.

Reply:
Current proposal can be viewed here:
https://github.com/maciekfabia/nrfxlib/pull/2/files

Reply:
> 
> 
> The reason is quite simple:
> The zephyr_library_* extension is mentioned and documented only in the CMake source code (at least - I cannot find it elsewhere). This makes it hard to find, understand and use. The only examples you may find is to go through other modules and try to find something similar to your needs. It would be great if you can point me to the tutorial on how to use zephyr_library_* extensions.

Agree, and I actually raised this 2 years ago: https://github.com/zephyrproject-rtos/zephyr/issues/9947
seems it was closed (resolved) without being properly resolved.

Reopened now.

> On the other hand - if you try to use "raw" CMake commands (fall-back to what you can find on the Internet), it will fail to link to the default zephyr target.
> 

I agree, which is why I raised this almost 2 years ago:
https://github.com/zephyrproject-rtos/zephyr/issues/8439

but until that is resolved, we have to stick to zephyr cmake wrappers.

Reply:
> 
> 
> Current proposal can be viewed here:
> https://github.com/maciekfabia/nrfxlib/pull/2/files

Good direction 👍 
Placed some comments in there.